### PR TITLE
Catch and fix more warnings in Java sources

### PR DESCRIPTION
**Pull request by ennru**
Enable more Java linting flags and﻿fix a few warnings detected by it.

### CODE REVIEW

**Comment by seglo** [comment]
Needs some `javafmtAll`

**Comment by seglo** [comment]
Doesn't like some of your flags, `-Xlint:removal` https://travis-ci.com/github/akka/alpakka/jobs/389305854#L575

**Comment by ennru** [comment]
Oh, I only tried it on JDK 11.

File: file/src/main/java/akka/stream/alpakka/file/impl/DirectoryChangesSource.java [hunk 1]
```diff
@@ -219,10 +219,9 @@ public String toString() {
    *     there was no changes, if the JDK implementation is slow, it will not help lowering this
    * @param maxBufferSize Maximum number of buffered directory changes before the stage fails
    */
-  @SuppressWarnings("unchecked")
   public static Source<Pair<Path, DirectoryChange>, NotUsed> create(
       Path directoryPath, FiniteDuration pollInterval, int maxBufferSize) {
     return Source.fromGraph(
-        new DirectoryChangesSource(directoryPath, pollInterval, maxBufferSize, Pair::apply));
+        new DirectoryChangesSource<>(directoryPath, pollInterval, maxBufferSize, Pair::apply));
```
Comment (raboof) [inline file/src/main/java/akka/stream/alpakka/file/impl/DirectoryChangesSource.java:225]: nice

**Review by raboof — APPROVED** [review]
neat

**Pull request merged**